rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -211,7 +232,49 @@ impl FirmwareProcessor {
                         txn.start_txn().send_response(&capabilities.to_bytes())?;
                         continue;
                     }
+
+                    CommandId::STASH_MEASUREMENT => {
+                        if measurement_count == MEASUREMENT_MAX_COUNT {
+                            cprintln!(
+                                "[afmc] Maximum supported number of measurements already received, ignoring."
+                            );
+                            txn.start_txn().complete(false)?;
+                            continue;
+                        }
+
+                        // Re-borrow mailbox to work around https://github.com/rust-lang/rust/issues/54663
+                        let txn = mbox
+                            .peek_recv()
+                            .ok_or(CaliptraError::FW_PROC_MAILBOX_STATE_INCONSISTENT)?;
+
+                        let mut txn = ManuallyDrop::new(txn.start_txn());
+                        Self::stash_measurement(
+                            pcr_bank,
+                            env.sha384,
+                            measurement_log,
+                            &mut txn,
+                            measurement_count,
+                        )?;
+                        measurement_count += 1;
+                        txn.complete(true)?;
+                    }
+
                     CommandId::FIRMWARE_LOAD => {
+                        // If no measurement was received, extend a well-known measurement.
+                        if measurement_count == 0 {
+                            let fake_measurement = StashMeasurement {
+                                measurement: [0xFFFFFFFF; 12],
+                                ..Default::default()
+                            };
+                            Self::extend_measurement(
+                                pcr_bank,
+                                env.sha384,
+                                measurement_log,
+                                &fake_measurement,
+                                0_usize,
+                            )?;
+                        }
```

Comment:
I assume this is here to defend against STASH_MEASUREMENT commands being dropped by an attacker?

Can we unconditionally do this? It cleanly separates whether measurements were made during ROM or RT execution and makes it a bit easier for verifiers to interpret.

Reply:
PR mostly LGTM, only open for me is this comment

Reply:
This was discussed with Varun/VishalS of having the fake measurement unconditionally and the consensus was to be either/or. We can re-visit this and fix it as a bug fix.